runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/FlinkStreamingTransformTranslators.java

```diff
@@ -284,8 +286,16 @@ public void flatMap(T s, Collector<WindowedValue<T>> collector) throws Exception
               }
             }).assignTimestampsAndWatermarks(new IngestionTimeExtractor<WindowedValue<T>>());
       } else {
-        source = context.getExecutionEnvironment()
-            .addSource(new UnboundedSourceWrapper<>(context.getPipelineOptions(), transform));
+        try {
+          UnboundedSourceWrapper<T, UnboundedSource.CheckpointMark> sourceWrapper =
+              new UnboundedSourceWrapper<>(
+                  context.getPipelineOptions(),
+                  (UnboundedSource) transform.getSource(),
```

Comment:
Yes, you can't avoid the casting because of unknown generics but I meant assigning `transform.getSource()` to a variable with a SupressWarnings annotation. This way you avoid declaring the whole method with SupressWarnings.